### ISSUE TITLE
Creating a rule to validate schema name casing.

### ISSUE THREAD

**Issue by miller79** [stoplightio/spectral#443]
**Chore summary**
I'm trying to create a rule that would validate schema name casing and am not able to see how to set the given and fields to do this.  Here was the sample I tried:

```yaml
rules:
  casing-rule:
    description: Casing Rule Testing
    recommended: true
    type: style
    given: "$..schemas"
    then:
      function: pattern
      functionOptions:
        match: ^[1][a-zA-Z]*$
```

**Additional context**
Any guidance that can be given to performing this or if this feature is not currently available as the documentation does not show this to be possible.

**Comment by miller79** [comment]
Figured out how to do it by doing the following:

```
  schema-names-pascal-case:
    description: Schema names MUST be written in PascalCase
    message: "{{property}} is not PascalCase: {{error}}"
    recommended: true
    type: style
    given: "$.components.schemas.*~"
    then:
      function: pattern
      functionOptions:
        match: ^[A-Z][a-zA-Z0-9]*$
```

Closing this out.

**Comment by philsturgeon** [comment]
Nicely done! If you'd like to get your hands dirty with actual code, we do have a case rule planned which has not been started yet. #405

**Comment by miller79** [comment]
I may do so we are wanting a linter for OpenAPI in my company and I’m owning it so definitely will be getting deep into things with it overtime so definitely will look into it.

**Comment by philsturgeon** [comment]
Fantastic! You’ve found your punter, just gotta make the rules now. We’ve got some docs: https://stoplight.io/p/docs/gh/stoplightio/spectral should help out
